rfc/rfc-0009-controller-unit-testing.md

```diff
@@ -0,0 +1,84 @@
+# RFC-0009: Controller Unit Testing
+
+**Authors:** Glyn Normington
+
+**Status:**
+
+**Pull Request URL:** https://github.com/projectriff/riff/pull/1372
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+The Kubernetes controllers in riff need unit testing in a consistent way. Other projects
+have adopted and promoted various ways of testing controllers. For ease of maintenance,
+riff needs to adopt a single, consistent approach.
+
+For the purpose of this document, unit testing refers to testing the
+`Reconcile` method of a controller.
+
+Apart from checking that various objects are created etc., it should be straigthtforward
+to test error paths by injecting errors in object lookup and creation.
+
+### Anti-Goals
+
+None.
+
+## Solution
+
+The approach taken is to use a fake client and other interfaces and introduce a test framework
```

Comment:
should other interfaces be strictly defined here?

Reply:
Probably too much detail. I've remove mention of other interfaces.